### ISSUE TITLE
Feature suggestion: Read cell contents in Excel when pressing CONTROL+` (toggle formula)

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#7043]
In Excel, pressing CONTROL+` toggles whether formulas or cell values are displayed visually in the cells themselves.

When this is pressed, NVDA does not announce anything, as the focus has not moved, however, it would be useful if the cell was read again, as this command could then be used to quickly read the formula of the current cell.

**Comment by SaschaCowley** [comment]
This seems to work in NVDA 2025.1 betas with Microsoft® Excel® for Microsoft 365 MSO (Version 2504 Build 16.0.18730.20122) 64-bit when using UIA in Excel, but not when using the object model (the default).
@CyrilleB79 this may be something you'd be interested in looking into

**Comment by CyrilleB79** [comment]
With NVDA 2025.1beta5 and a recent Office 2016 version (Microsoft® Outlook® 2016 MSO (Version 2503 Build 16.0.18623.20208) 32 bits):
* with Excel legacy mode (default), "Display formula" / "Hide formula" is reported.
* with Excel UIA mode (still experimental), the content of the cell is reported only if it has changed, and then "Display formula" / "Hide formula" is reported.

In both cases, errors are logged though.

With Narrator, what is reported is similar to NVDA's Excel UIA mode, i.e. cell content if it has changed and "Formula displayed/hidden".

With an older Office release (Office 2021 LTSC), nothing is reported.

So there are 2 issues:
* the errors seen in newer releases should be fixed (and maybe the cell's content could be thus also reported in legacy mode)
* maybe this feature could be back-ported on older Office releases (if we consider it's worth it).
